### PR TITLE
select-all checkbox, fixes #44

### LINKED ISSUE #44: Feature request: select all (visible) units in curation widget.
transferred from: https://github.com/flatironinstitute/labbox-ephys/issues/459

### PR DESCRIPTION
**Pull request by magland**

### CODE REVIEW

File: src/python/sortingview/gui/extensions/unitstable/Units/TableWidget.tsx [hunk 1]
```diff
@@ -188,7 +206,8 @@ const TableWidget: FunctionComponent<Props> = (props) => {
                         }
                         setSortFieldOrder(newSortFieldOrder)
                     }}
-                    onDeselectAll={selectedRowIds.length > 0 ? () => {onSelectedRowIdsChanged([])} : undefined}
+                    onDeselectAll={selectedRowIds.length === sortedRows.length ? handleDeselectAll : undefined}
```
Comment (magland) [inline src/python/sortingview/gui/extensions/unitstable/Units/TableWidget.tsx:209]: Show select-all checkbox if not everything is selected. Show deselect-all checkbox if everything is selected. So, to deselect all, you must first select all.

File: src/python/sortingview/gui/commonComponents/SortingUnitPairPlotGrid/SortingUnitPairPlotGrid.tsx [hunk 2]
```diff
@@ -12,6 +12,11 @@ type Props = {
 }
 
 const SortingUnitPairPlotGrid: FunctionComponent<Props> = ({ sorting, selection, selectionDispatch, unitIds, unitPairComponent, sortingSelector }) => {
+    const maxNumUnits = 6
+    if (unitIds.length > maxNumUnits) {
+        return <span>Please select {maxNumUnits} or fewer units</span>
+    }
+
```
Comment (magland) [inline src/python/sortingview/gui/commonComponents/SortingUnitPairPlotGrid/SortingUnitPairPlotGrid.tsx:19]: If we are going to allow select-all, then we better not allow more than 6 units to be selected for cross-correlograms.

**Review by jsoules — COMMENTED** [review]
Pushed a commit to make the (de)select-all button honor the lock state. (Bonus--I got to check that the `useLocalUnitIds` hook does actually work as designed and allow modification of its local state, assuming the UI isn't disabled!)

As I mentioned inline, it'd be nice to confirm how this is working when there's more than 6 units to select for the cross-correlograms. (The code looks right, but I didn't see any good data for a test, and I want to make sure we don't lock people out of deselecting in that view.)

Otherwise I think this is good to go.

**Comment by magland** [comment]
@jsoules nice catch wrt locking switch. I'll test out the >6 units case.

**Pull request merged**